service-base/src/main/java/org/eclipse/hono/service/http/DefaultFailureHandler.java

```diff
@@ -70,6 +70,7 @@ public void handle(final RoutingContext ctx) {
                         sendError(ctx.response(), HttpURLConnection.HTTP_INTERNAL_ERROR, ctx.failure().getMessage());
                     }
                 } else if (ctx.statusCode() != -1) {
+                    ctx.response().setStatusCode(ctx.statusCode());
```

Comment:
I do not think that we can/should fix this issue by trying to set the status code on the response like this.
The underlying problem is that we are (mis-)using the *ctx.response().setStatusMessage()* method together with *ctx.fail(int)* to convey detail information about a problem that has occurred while processing the request. However, we have a much more powerful tool at hand for doing this, which is using *ctx.fail(Throwable)* which we can invoke with a `ServiceInvocationException` containing the status code and detail message in one go. This is already handled correctly in the block above.
I therefore would suggest to simply pass `null` instead of *ctx.response().getStatusMessage()* to the *sendError()* method here and replace the erroneous usage of *ctx.fail(int)* in
* org.eclipse.hono.service.http.AbstractHttpEndpoint#extractRequiredJsonPayload
* org.eclipse.hono.service.credentials.CredentialsHttpEndpoint#updateCredentials
* org.eclipse.hono.service.tenant.TenantHttpEndpoint#checkPayloadForTenantId

with *ctx.fail(ClientErrorException)* ...